docs/html/user_guide.rst

```diff
@@ -1306,6 +1306,238 @@ issue tracker`_ if you believe that your problem has exposed a bug in pip.
 
 .. _`Using pip from your program`:
 
+Backtracking user guide information
+===================================
+
+Related to `GH issue 8713 <https://github.com/pypa/pip/issues/8713>`__.
+
+Objective
+---------
+
+The purpose of this content is to provide documentation of 1) why this
+occurs, and 2) how the user can possible solve it.
+
+Location
+--------
+
+The end location of this content will be on\ `the pip user
+guide <https://pip.pypa.io/en/latest/user_guide/#id35>`__, somewhere
+near the Resolution Impossible section.
+
+--------------
+
+Dependency resolution backtracking
+==================================
+
+Or more commonly known as *"Why does pip download multiple versions of
+the same package over and over again during an install?"*.
+
+The purpose of this section is to provide explanation of why
+backtracking happens, and practical suggestions to pip users who
+encounter dependency resolution backtracking during a ``pip install``
+command.
+
+What is backtracking?
+---------------------
+
+Backtracking is not a bug, or an unexpected behaviour. It is part of the
+way pip's dependency resolution process works.
+
+During a pip install (e.g. ``pip install tea``), pip needs to work out
+the packages dependencies (e.g. ``spoon``, ``hot-water``, ``cup`` etc),
+and the versions of each of these packages it needs to install.
+
+For each of these dependent packages, it needs to work out which version
+of a package is a good candidate to install.
+
+If everythig goes well, this will result in pip computing a set of
+compatible versions of all these packages.
+
+In the case where a package has a lot of versions, arriving at a good
+candidate can take a lot of time. (The amount of time depends on the
+package size, the number of versions pip must try, among other things)
+
+How does backtracking work?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pip does this by trying one version, if it finds out it isn’t compatible
+it needs to “go back” (backtrack) and download an older version.
+
+It then tries that version, if it is successful it will continue onto
+the next package - if not it will continue to backtrack until it finds a
+compatible version.
+
+At the end of this process, pip will have the set of compatible
+packages, which it then installs.
+
+This backtrack behaviour can end in 2 ways - either 1) it will
+successfully find a set packages it can install (good news!), or 2) it
+will eventually display `resolution
+impossible <https://pip.pypa.io/en/latest/user_guide/#id35>`__ error
+message (not so good).
+
+If pip starts backtracking during dependency resolution, it does not
+know how long it will backtrack, and how much computation would be
+needed.
+
+Why does backtracking occur?
+----------------------------
+
+With the release of the new resolver (resolver 2020), pip is now more
```

Comment:
```suggestion
With the release of the new resolver (:ref:`Resolver changes 2020`), pip is now more
```

This way you can link directly to the part of the documentation that explains the new resolver.